core/src/apps/thp/credential_manager.py

```diff
@@ -0,0 +1,108 @@
+from typing import TYPE_CHECKING
+
+from trezor import protobuf
+from trezor.crypto import hmac
+from trezor.messages import (
+    ThpAuthenticatedCredentialData,
+    ThpCredentialMetadata,
+    ThpPairingCredential,
+)
+from trezor.wire import wrap_protobuf_load
+
+if TYPE_CHECKING:
+    from apps.common.paths import Slip21Path
+
+
+def derive_cred_auth_key() -> bytes:
+    """
+    Derive current credential authentication mac-ing key from device secret.
+    """
+    from storage.device import get_cred_auth_key_counter, get_device_secret
+
+    from apps.common.seed import Slip21Node
+
+    # Derive the key using SLIP-21 https://github.com/satoshilabs/slips/blob/master/slip-0021.md,
+    # the derivation path is m/"Credential authentication key"/(counter 4-byte BE)
+
+    thp_secret = get_device_secret()
+    label = b"Credential authentication key"
+    counter = get_cred_auth_key_counter()
+    path: Slip21Path = [label, counter]
+
+    symmetric_key_node: Slip21Node = Slip21Node(thp_secret)
+    symmetric_key_node.derive_path(path)
+    cred_auth_key = symmetric_key_node.key()
+
+    return cred_auth_key
+
+
+def invalidate_cred_auth_key() -> None:
+    from storage.device import increment_cred_auth_key_counter
+
+    increment_cred_auth_key_counter()
+
+
+def issue_credential(
+    host_static_pubkey: bytes,
+    credential_metadata: ThpCredentialMetadata,
+) -> bytes:
+    """
+    Issue a pairing credential binded to the provided host static public key
+    and credential metadata.
+    """
+    return _issue_credential(
```

Comment:
Why aren't `_issue_credential()` and `_validate_credential()` inlined? I don't see them being used anywhere else.

Reply:
My idea was to have the !credential key derivation" decoupled from the "credential issuance/validation". As, you mentioned, it is not used anywhere else, so it is unnecessary. 

-> Inlined.